src/main.rs

```diff
@@ -258,7 +258,14 @@ fn run() -> Result<(), Box<::std::error::Error>> {
 
     if let Some(length) = matches
         .value_of("length")
-        .and_then(|n| n.parse::<u64>().ok())
+        .and_then(|n| {
+            if n.starts_with("0x") {
```

Comment:
We should probably also adapt the help-text to make users aware of this possibility.